src/OpenIddict.Core/Infrastructure/OpenIddictProvider.Authentication.cs

```diff
@@ -13,16 +13,113 @@
 using AspNet.Security.OpenIdConnect.Server;
 using JetBrains.Annotations;
 using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Authentication;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.WebUtilities;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
 
 namespace OpenIddict.Infrastructure {
     public partial class OpenIddictProvider<TUser, TApplication, TAuthorization, TScope, TToken> : OpenIdConnectServerProvider
         where TUser : class where TApplication : class where TAuthorization : class where TScope : class where TToken : class {
+        public override async Task ExtractAuthorizationRequest([NotNull] ExtractAuthorizationRequestContext context) {
+            var services = context.HttpContext.RequestServices.GetRequiredService<OpenIddictServices<TUser, TApplication, TAuthorization, TScope, TToken>>();
+
+            // Reject requests using the unsupported request parameter.
+            if (!string.IsNullOrEmpty(context.Request.GetParameter(OpenIdConnectConstants.Parameters.Request))) {
```

Comment:
These 2 checks used to be part of ASOS but were removed in beta6 to allow implementers to support `request` and `request_uri` (http://openid.net/specs/openid-connect-core-1_0.html#RequestUriParameter).

Since OpenIddict doesn't implement it, these checks are added to reject requests that use one of these parameters.